index.js

```diff
@@ -198,11 +199,30 @@ function compilerArgsFromOptions(options, emitWarning) {
   }));
 }
 
+// Tries to locate where the Elm binaries are. First looks at node_modules, then globally
+//
+function findElmBinaries() {
+  let localElmBin = path.resolve(path.join("node_modules", ".bin", "elm-make"));
+
+  return new Promise(function(resolve, reject) {
+    fs.access(localElmBin, function(err) {
+      if (!err) return resolve(localElmBin);
+
+      which('elm-make', function(err, path) {
+        if (err) return reject("Unable to find Elm anywhere!");
+
+        return resolve(path);
+      });
+    })
+  });
+}
+
 module.exports = {
   compile: compile,
   compileSync: compileSync,
   compileWorker: require("./worker.js")(compile),
   compileToString: compileToString,
   compileToStringSync: compileToStringSync,
-  findAllDependencies: findAllDependencies
+  findAllDependencies: findAllDependencies,
+  findElmBinaries: findElmBinaries
```

Comment:
Maybe this name is bad?

Reply:
how about `preferedElmBinaries` or `elmBinariesLocation`?

Reply:
Yeah, based on the name I'd be surprised to learn this was searching `node_modules` first... 🤔 

That said, I haven't thought of a better one. 😅